### PR TITLE
wgengine: don't diagnose iOS NWPathMonitor connection probe timeouts

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: wgengine/pendopen.go [hunk 1]
```diff
@@ -106,6 +108,17 @@ func (e *userspaceEngine) trackOpenPostFilterOut(pp *packet.Parsed, t *tstun.TUN
 	}
 
 	flow := flowtrack.Tuple{Src: pp.Src, Dst: pp.Dst}
+
+	// iOS likes to probe Apple IPs on all interfaces to check for connectivity.
+	// Don't start timers tracking those. They won't succeed anyway. Avoids log spam
+	// like:
+	//    open-conn-track: timeout opening (100.115.73.60:52501 => 17.125.252.5:443); no associated peer node
+	if runtime.GOOS == "ios" && flow.Dst.Port == 443 && !tsaddr.IsTailscaleIP(flow.Dst.IP) {
+		if _, ok := e.magicConn.PeerForIP(flow.Dst.IP); !ok {
```
Comment (crawshaw) [inline wgengine/pendopen.go:117]: Won't this catch on someone attempting to access a web server on a shared subnet?

Comment (bradfitz) [reply wgengine/pendopen.go:117]: no, because the `!ok`

Comment (bradfitz) [reply wgengine/pendopen.go:117]: we only skip the timer if it's port 443 on iOS going to a non-Tailscale IP for which we have no route from any peer

Comment (crawshaw) [reply wgengine/pendopen.go:117]: oh I misread that as `ok`

**Review by crawshaw — APPROVED** [review]

**Pull request merged**